modules/nf-core/diamond/blastp/tests/main.nf.test

```diff
@@ -0,0 +1,102 @@
+nextflow_process {
+
+    name "Test Process DIAMOND_BLASTP"
+    script "../main.nf"
+    process "DIAMOND_BLASTP"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "diamond"
+    tag "diamond/blastp"
+
+    setup {
+        run("DIAMOND_MAKEDB") {
+            script "../../makedb/main.nf"
+            process {
+                """
+                input[0] = [ [id:'test'], [ file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true) ] ]
+                """
+            }
+        }
+    }
+
+    test("Should search for protein hits against a DIAMOND db and return a tab separated output file of hits") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [ [id:'test'], file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true) ]
+                input[1] = DIAMOND_MAKEDB.out.db
+                input[2] = 'txt'
+                input[3] = 'qseqid qlen'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.txt },
+                { assert path(process.out.txt.get(0).get(1)).md5 == "8131b1afd717f3d5f2f2417c5b562e6e" },
```

Comment:
Also not entirely sure, if this is the best way to use nf-test instead of using a snapshot file. There was a discussion to include a lint test for the existence of a snapshot file.  https://nfcore.slack.com/archives/C049MBCEW06/p1699012061232069

Reply:
Snapshots are not working properly for me with conda. Testing a bit, but if they continue to randomly fail, I will switch back to this way of testing until I find a solution.